core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -222,34 +300,91 @@ private DriverContext createDriverContext(TaskContext taskContext)
         }
 
-        private PartitionedOutputBuffer createPartitionedBuffer(OutputBuffers buffers, DataSize dataSize)
+        private TestingPartitionedOutputBuffer createPartitionedBuffer(OutputBuffers buffers, DataSize dataSize)
         {
-            return new PartitionedOutputBuffer(
+            return new TestingPartitionedOutputBuffer(
                     "task-instance-id",
                     new StateMachine<>("bufferState", SCHEDULER, OPEN, TERMINAL_BUFFER_STATES),
                     buffers,
                     dataSize,
                     () -> new SimpleLocalMemoryContext(newSimpleAggregatedMemoryContext(), "test"),
                     SCHEDULER);
         }
+
+        private static class TestingPartitionedOutputBuffer
+                extends PartitionedOutputBuffer
+        {
+            public TestingPartitionedOutputBuffer(
+                    String taskInstanceId,
+                    StateMachine<BufferState> state,
+                    OutputBuffers outputBuffers,
+                    DataSize maxBufferSize,
+                    Supplier<LocalMemoryContext> systemMemoryContextSupplier,
+                    Executor notificationExecutor)
+            {
+                super(taskInstanceId, state, outputBuffers, maxBufferSize, systemMemoryContextSupplier, notificationExecutor);
+            }
+
+            // Use a dummy enqueue method to avoid OutOfMemory error
+            @Override
+            public void enqueue(int partitionNumber, List<SerializedPage> pages)
+            {
+            }
+        }
     }
 
-    public static void main(String[] args)
-            throws RunnerException
+    public static RowType withDefaultFieldNames(List<Type> types)
     {
-        // assure the benchmarks are valid before running
-        BenchmarkData data = new BenchmarkData();
-        new BenchmarkPartitionedOutputOperator().addPage(data);
+        List<RowType.Field> fields = new ArrayList<>();
+        for (int i = 0; i < types.size(); i++) {
+            fields.add(new RowType.Field(Optional.of("field" + i), types.get(i)));
+        }
+
+        fields = unmodifiableList(fields);
+        return RowType.from(fields);
+    }
+
+    static {
+        try {
+            List<TestType> types = List.of(
+                    TestType.BIGINT,
+                    TestType.DICTIONARY_BIGINT,
+                    TestType.RLE_BIGINT,
+                    TestType.LONG_DECIMAL,
+                    TestType.INTEGER,
+                    TestType.SMALLINT,
+                    TestType.BOOLEAN,
+                    TestType.VARCHAR,
+                    TestType.ARRAY_BIGINT);
+            // use multiple types to pollute the profile
+            types.forEach(type -> {
+                BenchmarkData data = new BenchmarkData();
+                data.type = type;
+                data.setup();
+                data.pageCount = 1;
+                new BenchmarkPartitionedOutputOperator().addPage(data);
```

Comment:
you should use same `BenchmarkPartitionedOutputOperator` instance for polluting it with data. Otherwise JIT might get clever (I've seen that)